File: Inventarios.Api/Inventarios.api.Test/Configuration/AuthHelper.cs
using System.Net.Http.Json;
using System.Text.Json;

namespace Inventarios.api.Test.Configuration;

public static class AuthHelper
{
    public static async Task<string> GetAdminTokenAsync(HttpClient http)
    {
        var request = new
        {
            email = "diana2@inventarios.com",
            password = "Admin@1234"
        };

        var response = await http.PostAsJsonAsync("/api/Auth/login", request);
        var body = await response.Content.ReadAsStringAsync();

        if (response.IsSuccessStatusCode)
        {
            var json = JsonSerializer.Deserialize<JsonElement>(body);
            var token = json
                .GetProperty("data")
                .GetProperty("token")
                .GetString();

            if (!string.IsNullOrEmpty(token))
                return token;
        }

        throw new InvalidOperationException(
            $"No se pudo obtener token. Status: {response.StatusCode}. Body: {body}");
    }
}

File: Inventarios.Api.Nest/productos.api/test/auth.helper.ts
import request from 'supertest';
import { INestApplication } from '@nestjs/common';

export async function getAdminToken(app: INestApplication): Promise<string> {
  const response = await request(app.getHttpServer())
    .post('/api/auth/login')
    .send({
      email: 'admin@productos.com',
      password: 'Admin@1234',
    });

  if (!response.body?.data?.token)
    throw new Error(
      `No se pudo obtener token. Status: ${response.status}. Body: ${JSON.stringify(response.body)}`,
    );

  return response.body.data.token;
}

export async function getEmpleadoToken(
  app: INestApplication,
  email: string,
  password: string,
): Promise<string> {
  const response = await request(app.getHttpServer())
    .post('/api/auth/login')
    .send({ email, password });

  if (!response.body?.data?.token)
    throw new Error('No se pudo obtener token de empleado.');

  return response.body.data.token;
}